src/modules/accommodations/controllers/accommodations.controller.ts

```diff
@@ -1,32 +1,61 @@
-import { AccommodationsService } from '../services/accommodations.service';
-import { Controller, Get, Post, Put, Delete } from '@nestjs/common';
+import { Controller, Post, Param, Body } from '@nestjs/common';
+import { AccommodationService } from '#/modules/accommodations/services/accommodations.service';
+import { AccommodationAddressService } from '#/modules/accommodations/services/accommodation-address.service';
+import { AccommodationImageService } from '#/modules/accommodations/services/accommodation-image.service';
+import { AccommodationAmenityService } from '#/modules/accommodations/services/accommodation-amenity.service';
+import { AccommodationAddressDto } from '#/modules/accommodations/dto/requests/create-accommodation-address.req';
+import { AccommodationImageDto } from '#/modules/accommodations/dto/requests/create-accommodation-image.req';
+import { AccommodationAmenityDto } from '#/modules/accommodations/dto/requests/create-accommodation-amenity.req';
+import { AccommodationAddress } from '#/modules/accommodations/entities/accommodation-address.entity';
+import { AccommodationImage } from '#/modules/accommodations/entities/accommodation-image.entity';
+import { AccommodationAmenity } from '#/modules/accommodations/entities/accommodation-amenity.entity';
+import { Accommodation } from '#/modules/accommodations/entities/accommodations.entity';
+import { AccommodationDto } from '#/modules/accommodations/dto/requests/create-accommodation.req';
 
-@Controller('accommodation')
-export class AccommodationsController {
-  constructor(private readonly AccommodationService: AccommodationsService) {}
+@Controller('accommodations')
+export class AccommodationController {
+  constructor(
+    private readonly accommodationService: AccommodationService,
+    private readonly accommodationAddressService: AccommodationAddressService,
+    private readonly accommodationImageService: AccommodationImageService,
+    private readonly accommodationAmenityService: AccommodationAmenityService,
+  ) {}
 
   @Post()
-  create() {
-    return this.AccommodationService.create();
+  async createAccommodation(
+    @Body() createAccommodationDto: AccommodationDto,
+  ): Promise<Accommodation> {
+    return this.accommodationService.create(createAccommodationDto);
   }
 
-  @Get()
-  findAll() {
-    return this.AccommodationService.findAll();
+  // Route to create an address for an accommodation
+  @Post(':accommodationId/address')
+  async createAddress(
```

Comment:
Not sure that we need this endpoint. Address is created when accommodation is created